tv-shows-app/src/components/features/shows/ShowContainer/ShowContainer.tsx

```diff
@@ -1,83 +1,100 @@
+import { CiWarning } from "react-icons/ci";
 
 export default function ShowContainer() {
-  const title = "Brooklyn Nine-Nine";
-  const description =
-    "Brooklyn Nine-Nine is a comedic TV series that follows the hilarious antics of Detective Jake Peralta and his diverse, quirky colleagues at the 99th precinct of the NYPD in Brooklyn, led by their stern but lovable Captain Holt.";
-  const imageUrl = "/assets/brooklyn99.jpg";
+  const { id } = useParams() as Params;
+  const {
+    data: showResponse,
+    error,
+    isValidating,
+  } = useSWR(`/api/shows/${id}`, () => getShow(id));
 
-  const mockReviewList: IReviewList = {
-    reviews: [],
-  };
-
-  const [reviewList, setReviewList] = useState(mockReviewList);
-  let averageRating = calculateAverage(reviewList);
+  const [reviewList, setReviewList] = useState<IReviewList>({ reviews: [] });
 
   useEffect(() => {
     const loadedList = loadFromLocalStorage();
     setReviewList(loadedList);
   }, []);
 
   function loadFromLocalStorage() {
-    const reviewListLoaded = localStorage.getItem("reviews");
+    const reviewListLoaded = localStorage.getItem("reviews" + id);
     if (!reviewListLoaded) {
-      return mockReviewList;
+      return { reviews: [] };
     }
     return JSON.parse(reviewListLoaded);
   }
 
-  function saveToLocalStorage(reviewList: IReviewList) {
-    localStorage.setItem("reviews", JSON.stringify(reviewList));
+  function saveToLocalStorage(reviewListToSave: IReviewList) {
+    localStorage.setItem("reviews" + id, JSON.stringify(reviewListToSave));
   }
 
   function addShowReview(review: IReview) {
-    const newList = {
-      reviews: [...reviewList.reviews, review],
-    };
+    const newList = { reviews: [...reviewList.reviews, review] };
     setReviewList(newList);
     saveToLocalStorage(newList);
   }
 
   function deleteShowReview(reviewToRemove: IReview) {
-    const newList: IReviewList = {
-      reviews: reviewList.reviews.filter((review) => review != reviewToRemove),
+    const newList = {
+      reviews: reviewList.reviews.filter((review) => review !== reviewToRemove),
     };
     setReviewList(newList);
     saveToLocalStorage(newList);
   }
 
-  function calculateAverage(reviewList: IReviewList) {
-    if (!reviewList.reviews.length) {
-      return 0;
-    }
-
-    let sum = 0;
-    reviewList.reviews.forEach((review) => (sum += review.rating));
+  if (isValidating) {
+    return (
+      <Flex margin="auto">
+        <Spinner boxSize={50} />
+      </Flex>
+    );
+  }
 
-    return sum / reviewList.reviews.length;
+  if (error) {
+    return (
+      <Flex margin="auto" justifyContent="center" alignItems="center" gap={5}>
+        {" "}
+        <CiWarning size={50} />
+        <Text letterSpacing="wide">Oops... Something went wrong</Text>
+      </Flex>
+    );
   }
 
+  const {
+    id: showId,
+    title,
+    description,
+    average_rating,
+    image_url,
+  } = showResponse ?? {};
```

Comment:
I can't speak for everyone, but in practice I much more often see `showResponse || {}`. The behaviour is not the same, but similar and I believe in your case it doesn't make a difference.
No need to change this, just FYI.